### PR TITLE
refactor: expose CSharpBaseExecutorProxy via _internal

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

This pull request primarily focuses on refactoring the code to improve the structure of the executor proxies in the `promptflow` package. The changes are mainly concentrated in the `src/promptflow/promptflow/batch/_base_executor_proxy.py` and `src/promptflow/promptflow/batch/_csharp_base_executor_proxy.py` files. The refactor includes the addition of new methods and properties, the removal of redundant code, and the reorganization of code between the base and derived executor proxy classes.

Here are the most important changes:

* [`src/promptflow/promptflow/_internal/__init__.py`](diffhunk://#diff-790bee87676597aac496dc18b945a811cf4591868dc9d601cf48d8654d6ec0e9R111): Imported `CSharpBaseExecutorProxy` from `promptflow.batch._csharp_base_executor_proxy` to make it available for use in the package.

Refactoring of executor proxies:

* [`src/promptflow/promptflow/batch/_base_executor_proxy.py`](diffhunk://#diff-44ac8eeb30a630bd24987e92f8bd5a180d57a3ad067db5d937561f21771e14c5R98-R153): Added the `__init__`, `working_dir`, `_get_flow_meta`, and `_get_tool_metadata` methods to the `APIBasedExecutorProxy` class. These changes allow the executor proxy to accept a working directory during initialization and provide methods to retrieve flow metadata and tool metadata from the working directory.
* [`src/promptflow/promptflow/batch/_csharp_base_executor_proxy.py`](diffhunk://#diff-2237f977aca2e9808fd5c2957c79d4be28b0525c02cdfd801c0160ad3315836dL22-R21): Removed redundant code from the `CSharpBaseExecutorProxy` class that was moved to the base class `APIBasedExecutorProxy`. This includes removing the `working_dir`, `_get_flow_meta`, and `_get_tool_metadata` methods, and modifying the `__init__` method to call the base class constructor. [[1]](diffhunk://#diff-2237f977aca2e9808fd5c2957c79d4be28b0525c02cdfd801c0160ad3315836dL22-R21) [[2]](diffhunk://#diff-2237f977aca2e9808fd5c2957c79d4be28b0525c02cdfd801c0160ad3315836dL82-L102)

Import changes:

* [`src/promptflow/promptflow/batch/_base_executor_proxy.py`](diffhunk://#diff-44ac8eeb30a630bd24987e92f8bd5a180d57a3ad067db5d937561f21771e14c5R6-R15): Imported `json`, `DEFAULT_ENCODING`, `MetaFileNotFound`, and `MetaFileReadError` to support the new methods added to the `APIBasedExecutorProxy` class.
* [`src/promptflow/promptflow/batch/_csharp_base_executor_proxy.py`](diffhunk://#diff-2237f977aca2e9808fd5c2957c79d4be28b0525c02cdfd801c0160ad3315836dL4-L9): Removed imports of `json`, `DEFAULT_ENCODING`, `FLOW_TOOLS_JSON`, `PROMPT_FLOW_DIR_NAME`, `MetaFileNotFound`, and `MetaFileReadError` that are no longer needed due to the code refactor.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/batch/_base_executor_proxy.py [hunk 1]
```diff
@@ -94,6 +95,62 @@ async def ensure_executor_health(self):
...
+    @property
+    def working_dir(self) -> Path:
+        """
+        The working directory of the executor, usually the flow directory,
+        where we can find metadata under .promptflow.
+        """
+        return self._working_dir
+
+    def _get_flow_meta(self) -> dict:
+        flow_meta_json_path = self.working_dir / ".promptflow" / "flow.json"
```
Comment (PeiwenGaoMS) [inline src/promptflow/promptflow/batch/_base_executor_proxy.py:120]: Also `PROMPT_FLOW_DIR_NAME `

Comment (elliotzh) [reply src/promptflow/promptflow/batch/_base_executor_proxy.py]: updated

**Review by PeiwenGaoMS — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/batch/_base_executor_proxy.py [hunk 2]
```diff
@@ -94,6 +96,60 @@ async def ensure_executor_health(self):
...
+    @property
+    def working_dir(self) -> Path:
+        """
+        The working directory of the executor, usually the flow directory,
+        where we can find metadata under .promptflow.
+        """
+        return self._working_dir
+
+    def _get_flow_meta(self) -> dict:
+        flow_meta_json_path = self.working_dir / PROMPT_FLOW_DIR_NAME / FLOW_META_JSON
```
Comment (D-W-) [inline src/promptflow/promptflow/batch/_base_executor_proxy.py:121]: When flow changed, when will the meta file be regenerated?

Comment (elliotzh) [reply src/promptflow/promptflow/batch/_base_executor_proxy.py:121]: proxy just expose an interface and the generation will be controlled by SDK

**Pull request merged**